### PR TITLE
[Backport v3.3-branch] ARC: SMP: fix IDU mask setup

### PR DESCRIPTION
**Pull request by zephyrbot**
Backport d4d0907ede61fceb1e2200ee2867b9a4371917f1~2..d4d0907ede61fceb1e2200ee2867b9a4371917f1 from #55956.

Fixes: #58397

### CODE REVIEW

**Review by stephanosio — APPROVED** [review]

**Review by ruuddw — APPROVED** [review]

**Review by abrodkin — APPROVED** [review]

**Comment by cfriedt** [comment]
We have approval from v3.3.0 RM and ARC maintainer, so I'll merge this since it's green.

**Pull request merged**
